src/selection.cpp

```diff
@@ -155,38 +160,104 @@ unsigned int selNumSelected(unsigned int player)
 	return count;
 }
 
-// ---------------------------------------------------------------------
-// sub-function - selects all units with same name as one passed in
-static void selNameSelect(char *droidName, unsigned int player, bool bOnScreen)
+// Helper function to check whether the component stats of a unit can be found
+// in the combinations vector and, optionally, to add them to it if not
+static bool componentsInCombinations(DROID *psDroid, bool add)
 {
-	for (DROID *psDroid = apsDroidLists[player]; psDroid; psDroid = psDroid->psNext)
+	std::vector<uint32_t> components;
+	uint32_t stat;
+
+	// stats are sorted by their estimated usefulness to distinguish units:
+	// * the first weapon turret is the most common difference between them
+	//   followed by propulsions and bodies
+	// * the second weapon turret is least important because Hydras are rare
+	for (int c = 0; c < DROID_MAXCOMP + 2; c++)
 	{
-		if (!psDroid->selected)
+		switch(c)
 		{
-			if (!bOnScreen || droidOnScreen(psDroid, 0))
-			{
-				if (!strcmp(droidName, psDroid->aName))
-				{
-					SelectDroid(psDroid);
-				}
-			}
+			case 0: stat = psDroid->asWeaps[1].nStat; break;
+			case 1: stat = psDroid->asBits[COMP_ECM]; break;
+			case 2: stat = psDroid->asBits[COMP_BRAIN]; break;
+			case 3: stat = psDroid->asBits[COMP_SENSOR]; break;
+			case 4: stat = psDroid->asBits[COMP_REPAIRUNIT]; break;
+			case 5: stat = psDroid->asBits[COMP_CONSTRUCT]; break;
+			case 6: stat = psDroid->asBits[COMP_BODY]; break;
+			case 7: stat = psDroid->asBits[COMP_PROPULSION]; break;
+			case 8: stat = psDroid->asWeaps[0].nStat; break;
 		}
+
+		// keep the list of components short by not adding stats with
+		// the value 0 to its end, since they are redundant
+		if (!(stat == 0 && components.empty()))
```

Comment:
Build error on Travis:
`/Users/travis/build/Warzone2100/warzone2100/src/selection.cpp:191:9: variable 'stat' may be uninitialized when used here [-Werror,-Wconditional-uninitialized]`

Probably should initialize at the top of the function:
```diff
- uint32_t stat;
+ uint32_t stat = 0;
```

Reply:
Fixed.